internal/app/knitgateway/version.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package knitgateway
+
+import (
+	"context"
+	"fmt"
+
+	"connectrpc.com/connect"
+)
+
+const buildVersion = "v0.1.0-dev"
+
+//nolint:gochecknoglobals
+var (
+	// NB: This is a var instead of a const so it can be changed via -X ldflags.
+	buildVersionSuffix = ""
+)
+
+func Version() string {
+	return buildVersion + buildVersionSuffix
+}
+
+func UserAgentInterceptor(call connect.UnaryFunc) connect.UnaryFunc {
+	return func(ctx context.Context, req connect.AnyRequest) (connect.AnyResponse, error) {
+		// decorate user-agent with the program name and version
+		userAgent := fmt.Sprintf("knitgateway/%s %s", Version(), req.Header().Get("User-Agent"))
```

Comment:
Is there a convention for combining or forwarding User-Agent values?

I've seen the forwarded (upstream) user-agent wrapped in parentheses for easier parsing, but curious if you know otherwise.

Reply:
Hmmm, great question. I _thought_ this was the correct way to do it. However, [these docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent) suggest I got it backwards, and that extensions should go after, not before.

The default user agent for Connect is in the form `connect-go/1.6.0 (go1.20.3)`. (That's for the connect protocol; for gRPC, it instead uses `grpc-go-connect/1.6.0 (go1.20.3)`, which is in the form recommended by [the gRPC spec](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#user-agents).

Reply:
I just pushed a commit to reverse the order here, and put the knitgateway info at the end instead of the beginning.

Reply:
I think this makes sense. I've also noticed that "typically" the first item is the primary user-agent for the current request and subsequent ones are supplementary.

Thanks for adding this.